ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -487,4 +492,31 @@ private boolean isEligibleForCompaction(CompactionInfo ci, ShowCompactResponse c
     }
     return true;
   }
+
+  @VisibleForTesting
+  protected static void updateCompactionMetrics(ShowCompactResponse showCompactResponse) {
+    Map<String, ShowCompactResponseElement> lastElements = new HashMap<>();
+
+    // Get the last compaction for each db/table/partition
+    for(ShowCompactResponseElement element : showCompactResponse.getCompacts()) {
+      String key = element.getDbname() + "/" + element.getTablename() + "/" + element.getPartitionname();
```

Comment:
element.getPartitionname() can return null, resulting in a key="dbName/tableName/null"

Reply:
Added null check as suggested.
Thanks!